pulp_smash/tests/ostree/api_v2/test_sync_publish.py

```diff
@@ -0,0 +1,447 @@
+                    self.assertEqual(
+                        len(action['error_details']),
+                        0,
+                        task_body
+                    )
+
+
+class SyncWithoutFeedTestCase(_BaseTestCase):
+    """Without provided feed, repo synchronisation fails."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create repository without feed and run sync on it."""
+        super(SyncWithoutFeedTestCase, cls).setUpClass()
+        body = _gen_ostree_repo_body()
+        client = api.Client(cls.cfg, api.json_handler)
+        repo = client.post(REPOSITORY_PATH, body)
+        client.response_handler = api.echo_handler
+        cls.sync_report = client.post(
+            urljoin(repo['_href'], 'actions/sync/'),
+            {'override_config': {}}
+        )
+        cls.sync_report.raise_for_status()
+        cls.task_bodies = tuple(
+            utils.poll_spawned_tasks(cls.cfg, cls.sync_report.json()))
+        client.response_handler = api.json_handler
+        cls.resources.add(repo['_href'])
+
+    def test_start_sync_code(self):
+        """Assert the call to sync a repository returns an HTTP 202."""
+        self.assertEqual(self.sync_report.status_code, 202)
+
+    def test_task_error(self):
+        """Assert each task's "error" field is non-null."""
+        for i, task_body in enumerate(self.task_bodies):
+            with self.subTest(i=i):
+                self.assertIsNotNone(task_body['error'])
+
+    def test_task_traceback(self):
+        """Assert each task's "traceback" field is non-null."""
+        for i, task_body in enumerate(self.task_bodies):
+            with self.subTest(i=i):
+                self.assertIsNotNone(task_body['traceback'])
+
+    def test_number_tasks(self):
+        """Assert that two task were spawned."""
+        self.assertEqual(len(self.task_bodies), 1)
+
+
+class SyncInvalidFeedTestCase(_BaseTestCase):
+    """With invalid feed or branch, the sync completes with reported errors."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create an OSTree repository with an invalid feed or branch, sync."""
+        super(SyncInvalidFeedTestCase, cls).setUpClass()
+        client = api.Client(cls.cfg, api.json_handler)
+        bodies = tuple(_gen_ostree_repo_body() for _ in range(2))
+        bodies[0]['importer_config']['feed'] = utils.uuid4()  # invalid feed
+        bodies[0]['importer_config']['branches'] = [_VALID_BRANCHES[0]]
+        bodies[1]['importer_config']['feed'] = _VALID_FEED
+        bodies[1]['importer_config']['branches'] = [utils.uuid4()]
+        repos = tuple(client.post(REPOSITORY_PATH, body) for body in bodies)
+        client.response_handler = api.echo_handler
+        cls.sync_reports = []
+        for repo in repos:
+            cls.sync_reports.append(client.post(
+                urljoin(repo['_href'], 'actions/sync/'),
+                {'override_config': {}}
+            ))
+            cls.sync_reports[-1].raise_for_status()
```

Comment:
I don't see the logic behind setting the client's response handler to `echo_handler`, then calling `raise_for_status()` on each response. This should be equivalent:

```python
client.response_handler = api.safe_handler
cls.sync_reports = [
    client.post(...)
    for repo in repos
)
```